### PR TITLE
Circle db

### PR DESCRIPTION
**Pull request by Sriep**
Needed for https://github.com/loomnetwork/loomchain/pull/364

### CODE REVIEW

File: builtin/types/config/config.proto [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+message ConfigInitRequest {
+    Address oracle = 1;
+    repeated KeyValue settings = 2;
+}
+
+message Key {
+    string key = 1;
+}
+
+message KeyValue {
```
Comment (mattkanwisher) [inline builtin/types/config/config.proto]: maybe Setting would be a better name

File: builtin/types/config/config.proto [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    CHAIN = 0;
+    LEVELDB = 1;
+}
+
+message ConfigInitRequest {
+    Address oracle = 1;
+    repeated KeyValue settings = 2;
+}
+
+message Key {
```
Comment (mattkanwisher) [inline builtin/types/config/config.proto]: not sure this needs its own message

File: builtin/types/config/config.proto [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    string key = 1;
+}
+
+message KeyValue {
+    string key = 1;
+    Value value = 2;
+}
+
+message SetKeyValue {
```
Comment (mattkanwisher) [inline builtin/types/config/config.proto]: SetSetting or updateSetting would be better

File: builtin/types/config/config.proto (hunk 2 above)
Comment (mattkanwisher) [reply builtin/types/config/config.proto]: unless this is supposed to be GetKey ? or GetSetting?

Comment (Sriep) [reply builtin/types/config/config.proto]: It's the input parameter to the Get method, `(c *Config) Get(ctx contractpb.StaticContext, key ctypes.Key ) (*ctypes.Value, error)`

Comment (Sriep) [reply builtin/types/config/config.proto]: Maybe should be changed to RetrieveSetting or similar.

Comment (mattkanwisher) [reply builtin/types/config/config.proto]: GetSetting

File: builtin/types/config/config.proto [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    string key = 1;
+}
+
+message Setting {
+    string key = 1;
+    Value value = 2;
+}
+
+message UpdateSetting {
+    Address oracle = 1;
```
Comment (mattkanwisher) [inline builtin/types/config/config.proto]: lets remove this oracle

Comment (mattkanwisher) [reply builtin/types/config/config.proto]: I just removed this address for oracle

**Pull request merged**